tools/server/server.cpp

```diff
@@ -2515,6 +2557,41 @@ struct server_context {
         queue_results.send(std::move(res));
     }
 
+    void send_progress_response(server_slot & slot) {
+        // Only send progress if explicitly requested and streaming is enabled
+        if (!slot.params.return_progress || !slot.params.stream) {
+            return;
+        }
+
+        // Calculate current progress percentage
+        float current_progress = slot.n_prompt_tokens > 0 ?
+                                (float) slot.n_prompt_tokens_processed / slot.n_prompt_tokens : 0.0f;
```

Comment:
for example, if `n_prompt_tokens` is 10 but 6 of them are already in cache, then `n_prompt_tokens_processed` can only be ranging from 0 to 4